accounting/accounting_cli_functions.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+###############################################################
+# Copyright 2020 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+import sqlite3
+import argparse
+import time
+import sys
+
+import pandas as pd
+
+def view_user(user):
+    try:
+        # open connection to database
+        conn = sqlite3.connect("FluxAccounting.db")
```

Comment:
Rather than having a connection created at the start of every function (and having the database name/path hardcoded into each function), maybe it makes sense to add `conn` as an argument to these functions?  That would potentially enable future users of this code to perform multiple operations with a single connection, and it also means you only have to get the opening/closing logic correct once in `accounting_cli.py` rather than getting it correct multiple times in this file.

Reply:
Sorry, I think I might need a little more clarification 😬 do you mean specifying the path to the .db file as an additional argument on the command line? as an example:

`flux-accounting accounting/FluxAccounting.db -u fluxuser`

Reply:
With some help from @chu11, I think I was able to make this change. All of the functions in `accounting_cli_functions.py` now take a **conn** parameter that's passed in from `accounting_cli.py`:

```
def main():

    conn = sqlite3.connect("FluxAccounting.db")

    ... parse args and stuff ...

    if args.view_user:
        aclif.view_user(conn, args.view_user)
    elif args.add_user:
        aclif.add_user(conn, args.add_user)
    elif args.delete_user:
        aclif.delete_user(conn, args.delete_user)
    elif args.edit_user:
        aclif.edit_user(conn, args.edit_user)

    conn.close()
```

This should only establish one connection while it runs, and then closes the connection when it exits. I hope I understood you correctly! Sorry in advance if it wasn't.